### PR TITLE
Add static analysis action

### PR DESCRIPTION
**Pull request by elenadimitrova**
**Description**
Setup python and install [slither](https://github.com/crytic/slither/wiki/Usage) in a new github action to enable running static analysis as part of CI. Note that currently failures do not fail the github action.

To run `slither` locally in ./packages/contracts do
```
pip3 install slither-analyzer
yarn test:slither
```

Note that for performance the above relies on contracts to have been compiled before slither is run so ensure you've run `yarn build` beforehand.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,6 +14,9 @@ cache-ovm
 l2geth/build/bin
 packages/contracts/deployments/custom
 packages/contracts/coverage*
+packages/contracts/@ens*
+packages/contracts/@openzeppelin*
+packages/contracts/hardhat*
```
Comment (maurelian) [inline .gitignore:19]: Why do we need these? 

edit: OK, I see below.

File: packages/contracts/package.json [hunk 2]
```diff
@@ -32,6 +32,9 @@
     "test:contracts": "hardhat test --show-stack-traces",
     "test:gas": "hardhat test \"test/contracts/OVM/execution/OVM_StateManager.gas-spec.ts\" --no-compile --show-stack-traces",
     "test:coverage": "NODE_OPTIONS=--max_old_space_size=8192 hardhat coverage",
+    "test:slither": "slither .",
+    "pretest:slither": "rm -f @openzeppelin && rm -f @ens && rm -f hardhat && ln -s ../../node_modules/@openzeppelin @openzeppelin && ln -s ../../node_modules/@ens @ens && ln -s ../../node_modules/hardhat hardhat",
+    "posttest:slither": "rm -f @openzeppelin && rm -f @ens && rm -f hardhat",
```
Comment (maurelian) [inline packages/contracts/package.json:37]: Should we have a command that combines all of these steps into one?

**Review by maurelian — CHANGES_REQUESTED** [review]
I'm excited for this!
Requests: 
- Update readme for the package with the instructions from the PR description. 
- I'm also curious about the intended workflow, is the idea that you'd run `yarn pretest:slither`, then run `yarn test:slither` several times will doing fixes, and then finally run `yarn posttest:slither`? Hence my question in the review comments about having a combined command.

File: packages/contracts/package.json (hunk 2 above)
Comment (elenadimitrova) [reply packages/contracts/package.json:37]: These "pre" and "post" hooks are an npm feature and execute automatically before and after the command. There's nothing more we need to do. I've had to link these packages that way as slither does not work well in a monorepo setup and fails to read node modules from the root.

Comment (maurelian) [reply packages/contracts/package.json:37]: Oh cool! I didn't realize the hooks would get picked up even with the `:` in them.

**Review by maurelian — APPROVED** [review]
LGTM!

**Pull request merged**